test: Only wait for one operator instance to be ready

jrajahalme opened the pull request:
Waiting for only one Cilium operator instance to be ready allows
testing in single node environments, such as Minikube.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

aditighag commented on test/k8sT/assertionHelpers.go:
```diff
@@ -58,7 +58,7 @@ func ExpectCiliumNotRunning(vm *helpers.Kubectl) {
 func ExpectCiliumOperatorReady(vm *helpers.Kubectl) {
 	By("Waiting for cilium-operator to be ready")
-	err := vm.WaitforPods(helpers.CiliumNamespace, "-l name=cilium-operator", longTimeout)
+	err := vm.WaitForSinglePod(helpers.CiliumNamespace, "-l name=cilium-operator", longTimeout)
```
Shouldn't we check if it's a single node deployment, and call `WaitForSinglePod`/`WaitForPods` accordingly?

jrajahalme replied:
Maybe, but isn't the whole point of having two operator instances HA, i.e., having one of them operational should be fine? Unless we explicitly want to test that in a specific scenario both of them come up?

christarazi replied:
It's possible that we make a change in the future where the second operator instance fails and this wouldn't alert anyone that something unexpected happened. I think I would agree with Aditi.

nebril replied:
If we make any features that require operator to have two running instances for some reason, it should be tested separately as a separate e2e suite. I think this change is fine as it is.

jrajahalme replied:
Changed to retain the old behavior if number of k8s nodes >= 2.

aditighag reviewed (dismissed):
Thanks for this PR.

jrajahalme committed:
test: Clarify that WaitForSinglePod takes a name of a single pod rather than a filter.

WaitForSinglePod takes a name of a single pod so that the kubectl json
output can be unmarshaled in to v1.Pod rather than into a
v1.PodList. Change the parameter 'filter' into 'podname' and fix
comments.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

jrajahalme committed:
test: Only wait for one operator instance to be ready

Waiting for only one Cilium operator instance to be ready allows
testing in single node environments, such as Minikube.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>
